lua/nvim-tree/view.lua

```diff
@@ -202,12 +202,23 @@ function M.focus(winnr, open_if_closed)
   a.nvim_set_current_win(wnr)
 end
 
+local function get_width()
+  if type(M.View.width) == "string" then
+    if vim.endswith(M.View.width, "%") then
+      local width_as_number = tonumber(M.View.width:match('[^%d]*([%d%.]+)%%'))
+      local percent_as_decimal = width_as_number / 100
+      return math.floor(vim.o.columns * percent_as_decimal)
+    end
+  end
+  return M.View.width
```

Comment:
when you have a logic that makes the whole code indented this way, it's better to do it in the other way:
```lua
if type(M.View.width) == "number" then
  return M.View.width
end
...rest of the code
```

Reply:
Going to leave this open just in case: I may have not understood this fully. Please check the most recent commit!